app/services/person_constructor.rb

```diff
@@ -72,6 +73,12 @@ def strip_whitespace_from_values
     params.merge!(params){ |k, v, _| v.try(:strip) || v }
   end
 
+  def normalize_zip_code
+    if params[:zip_code]
+      params[:zip_code] = params[:zip_code].gsub!(/\A(\d{5})[^\w]?(\d{4})?\z/, '\1')
```

Comment:
Yeah, now that I think about this it could be a lot easier.
I was trying to do 3 things:
 - make sure that there's a `zip_code` param
 - grab the first 5 digits if it's a valid zip
 - nullify it if it's not a valid zip

Now I realize that the last thing was wrong: if it's not a valid zip, it should just pass it through to Location to be validated like normal.

I still want to make sure that it's a valid zip before grabbing the first 5, though. I wan't to reject 6-digit zips, etc